.github/workflows/ci.yml

```diff
@@ -1,17 +1,20 @@
-name: ci
-on: push
+name: CI
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, edited, synchronize, reopened]
+  push:
+    branches: [main]
 jobs:
-  ci:
+  CI:
     runs-on: ubuntu-latest
+    name: CI
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
           node-version-file: '.nvmrc'
+          cache: 'npm'
       - run: npm ci
       - run: npm run build
-      - run: npm run test
-      - if: startsWith(github.ref, 'refs/tags/v')
-        run: npm publish
-        env:
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
Could it be simpler to keep these lines instead of creating a `cd` workflow and duplicating the build steps? Or are we gaining certain benefits from separating `ci` and `cd`?

Reply:
The gain is the environment essentially, ergo the protection of the NPM token.